### PR TITLE
improvement(LOC-320): add text support to Spinner, use onRequestClose prop in FlyModal

### PR DESCRIPTION
**Pull request by claygriffiths**
**Summary:** Add text/children support for `<Spinner />`, use `onRequestClose` prop on `<FlyModal />` if provided. Some SCSS was also removed that should've been in Local itself.

----

### Screenshot

screen shot 2019-02-06 at 9 54 15 am

### CODE REVIEW

File: src/components/FlyModal/FlyModal.tsx [hunk 1]
```diff
@@ -33,7 +33,7 @@ interface IProps extends IReactComponentProps {
 	closeTimeoutMS?: number;
 	hasIcon?: boolean;
 	isOpen?: boolean;
-	onRequestClose?: () => void;
+	onRequestClose?: () => boolean | void;
```
Comment (flycrum) [inline src/components/FlyModal/FlyModal.tsx]: Just curious, why is this boolean or void?

File: src/components/Spinner/README.md [hunk 2]
```diff
@@ -1,4 +1,11 @@
-Basic spinner.
+Spinner without text.
 ```js
 <Spinner />
 ```
+
+Spinner with text.
+```js
+<Spinner>
+	With some text...
```
Comment (flycrum) [inline src/components/Spinner/README.md]: I love seeing this proved out in Styleguidist! 🙌

File: src/components/Spinner/Spinner.tsx [hunk 3]
```diff
@@ -5,10 +5,8 @@ import SpinnerSVG from '../../svg/spinner';
 import * as styles from './Spinner.sass';
 
 interface IProps extends IReactComponentProps {
-
-	className?: string;
-	color?: 'Gray25' | 'GrayDark50';
-
+	className?: string
```
Comment (flycrum) [inline src/components/Spinner/Spinner.tsx]: Yeah, I'm cool with ditching these extra lines. If we still want to keep them for classes, I think that's fine, but I'd love to ditch them for everything else.

File: src/styles/containers/SiteInfo.scss [hunk 4]
```diff
@@ -333,28 +333,4 @@
 			fill: $gray75;
 		}
 	}
-
```
Comment (flycrum) [inline src/styles/containers/SiteInfo.scss]: Bye bye Local-specific styles! 👋

File: src/components/FlyModal/FlyModal.tsx [hunk 5]
```diff
@@ -73,7 +81,7 @@ export default class FlyModal extends React.Component<IProps> {
 					className={classnames({
 						[styles.FlyModal__HasIcon]: this.props.hasIcon,
 					})}
-					onClick={FlyModal.onRequestClose}
+					onClick={this.onRequestClose.bind(this)}
```
Comment (flycrum) [inline src/components/FlyModal/FlyModal.tsx]: This is the only thing I'd really like to see changed. I'd prefer avoid ever using bind within the `render` function as it some undesirable re-rendering implications.

File: src/components/FlyModal/FlyModal.tsx (hunk 1 above)
Comment (claygriffiths) [reply src/components/FlyModal/FlyModal.tsx]: Good catch. I initially made it so returning false on `onRequestClose` would stop the modal from closing but I'm going to ditch that for now.

Comment (flycrum) [reply src/components/FlyModal/FlyModal.tsx]: That makes sense. I wondered cause I've done the same thing when building modal components in the past. Just didn't see it being used and didn't know if there were plans to use it that way later on.

File: src/components/Spinner/Spinner.tsx (hunk 3 above)
Comment (claygriffiths) [reply src/components/Spinner/Spinner.tsx]: Yeah, we really need to get some TSLint rules in place like we did with ESLint. It's a shame there isn't an easy way to port the settings.

**Pull request merged**